### ISSUE TITLE
Admission controller: Namespace and controller information lost before validation

### ISSUE THREAD

**Issue by HansK-p** [FairwindsOps/polaris#465]
### Installation Process
Polaris was installed using the Helm chart 3.0.0, with a docker image based on the master branch and with the following values.yaml:
<pre>
image:
  repository: k8sfaith/polaris
  tag: "2020-12-29"
  pullPolicy: Always

webhook:
  enable: true

config:
  checks:
    privilegeEscalationAllowed: danger

  exemptions:
    - namespace: ingress-nginx
      controllerNames:
        - ingress-nginx-controller
      containerNames:
        - controller
      rules:
        - privilegeEscalationAllowed
        - insecureCapabilities
        - notReadOnlyRootFilesystem

    - controllerNames:
        - ingress-nginx-controller
      containerNames:
        - controller
      rules:
        - privilegeEscalationAllowed
        - insecureCapabilities
        - notReadOnlyRootFilesystem

#    - containerNames:
#        - controller
#      rules:
#        - privilegeEscalationAllowed
#        - insecureCapabilities
#        - notReadOnlyRootFilesystem
</pre>

The deployment is installed using the following yaml file:
<pre>
---
apiVersion: v1
kind: Namespace
metadata:
  labels:
    name: ingress-nginx
  name: ingress-nginx
---
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    app.kubernetes.io/name: ingress-nginx
  name: ingress-nginx-controller
  namespace: ingress-nginx
spec:
  selector:
    matchLabels:
      app.kubernetes.io/name: ingress-nginx
  replicas: 1
  template:
    metadata:
      labels:
        app.kubernetes.io/name: ingress-nginx
    spec:
      containers:
        - name: controller
          image: "k8s.gcr.io/ingress-nginx/controller:v0.42.0@sha256:f7187418c647af4a0039938b0ab36c2322ac3662d16be69f9cc178bfd25f7eee"
          imagePullPolicy: IfNotPresent
          securityContext:
            capabilities:
                drop:
                - ALL
                add:
                - NET_BIND_SERVICE
            runAsUser: 101
            allowPrivilegeEscalation: true
          livenessProbe:
            httpGet:
              path: /healthz
              port: 10254
              scheme: HTTP
          readinessProbe:
            httpGet:
              path: /healthz
              port: 10254
              scheme: HTTP
          resources: 
            requests:
              cpu: 100m
              memory: 90Mi
</pre>

### Polaris Version
Polaris build from the master branch on 2020-12-29. Polaris v.3.0.0. didn't allow the deployment even with the modified rules file.

### Expected Behavior
The Polaris Admission controller should allow the deployment when controller name and/or namespace is specified in addition to container name.

### Actual Behavior
The Polaris Admission controller denied the deployment unless only container name was specified.

### Steps to Reproduce
1. Install cert-manager as a prerequisite
2. Create values.yaml as can be seen above
3. `kubectl create ns polaris`
4. `helm install -f values.yaml --namespace polaris test fairwinds-stable/polaris`
5. Wait for the Polaris installation
6. Create the file "deployment.yaml" as can be seen above
7. `kubectl apply -f deployment.yaml`
8. The deployment is denied with a message "Container controller: Privilege escalation should not be allowed" 
9. Uncomment the last part of values.yaml
10. `helm upgrade -f values.yaml --namespace polaris test fairwinds-stable/polaris`
11. Wait for the upgrade to finish
12. `kubectl apply -f deployment.yaml`
13. The deployment is allowed. 
14. `kubectl delete ns ingress-nginx`
15. Remove the image part of values.yaml
16. `helm upgrade -f values.yaml --namespace polaris test fairwinds-stable/polaris`
17. Wait for the upgrade to finish
18. `kubectl apply -f deployment.yaml`
19. The deployment is not allowed

### Additional Context
This has been tested on Kubernetes v.1.19.3 and v.1.20.1.

**Comment by baderbuddy** [comment]
Hi Hans, the container level exemptions are not in 3.0.0, which is why you saw the different behavior. But that definitely looks like an issue. I'll take a look into this when I get a chance

**Comment by HansK-p** [comment]
Hello
It actually looks like pod, controller and name space exemptions are working with a small change to the code. But there migt be side effects I'm not aware of.

See PR.

I've been testing a bit with the config below, and it looks like the admission controller filters on all options with the small change (namespace, controller name and container name):
<pre>
    - namespace: ingress-nginx
      controllerNames:
        - ingress-nginx-controller
      containerNames:
        - controllers
      rules:
        - privilegeEscalationAllowed
        - insecureCapabilities
        - notReadOnlyRootFilesystem
</pre>

Br
Hans K.

**Comment by gmolkvk** [comment]
I believe this may be related. I cannot seem to get exemptions working in 3.x. I don't use helm for my deployment, but the result should be the same.

My config.yaml
```
checks:
  hostNetworkSet: danger
exemptions:
- namespace: grafana-test
  controllerNames:
  - grafana
  containerNames:
  - grafana
  rules:
  - hostNetworkSet
```

No matter what combination of namespace, controllerNames, containerNames, or rules I put into the config, the exemption is not working. When I edit my grafana deployment in the grafana-test namespace and set the Deployment.spec.template.space.hostNetwork to true, the Polaris webhook will block the change:
```
admission webhook "polaris-webhook" denied the request: Polaris prevented this deployment due to configuration problems: - Pod: Host network should not be configured
```

The webhook logs leads me to believe this is related to this issue; because the namespace and pod name are also missing here:
```
time="2021-02-25T14:59:59Z" level=info msg="Starting request"
time="2021-02-25T14:59:59Z" level=info msg="1 validation errors found when validating "
```

**Comment by rbren** [comment]
This should be fixed in 3.1.4 - if y'all are still seeing it please let us know!

**Comment by gmolkvk** [comment]
I can verify that the exemptions work now
